### PR TITLE
Add workflows for releasing using the new method

### PR DESCRIPTION
**Pull request by pdewilde**
Pattern copied from abcxyz/pkg

Will create RELEASING.md in a future PR.

### CODE REVIEW

File: .github/workflows/draft-release.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    runs-on: 'ubuntu-latest'
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+    - name: 'Increment version'
+      id: 'increment-version'
+      uses: 'abcxyz/pkg/.github/actions/increment-version@main' # ratchet:exclude
+      with:
+        version_strategy: '${{ inputs.version_strategy }}'
```
Comment (verbanicm) [inline .github/workflows/draft-release.yml:46]: You should add a VERSION file with 0.0.0 as its contents just to make sure its there for when you release.

File: .github/workflows/draft-release.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    - name: 'Mint token'
```
Comment (verbanicm) [inline .github/workflows/draft-release.yml:48]: Make sure you have an entry for token minter for this.

File: .github/workflows/release.yml
Comment (verbanicm) [inline .github/workflows/release.yml:1]: I this from JVS? I thought we didnt have maven or binaries for this?

**Comment by drevell** [comment]
Removed myself as reviewer because I don't know about the new release process and it seems Mike is reviewing already.

File: .github/workflows/release.yml
Comment (pdewilde) [reply .github/workflows/release.yml:1]: Good catch. Its from pkg, checked to ensure it wasn't doing containers, but missed the other parts. I'll remove

File: .github/workflows/draft-release.yml (hunk 2 above)
Comment (pdewilde) [reply .github/workflows/draft-release.yml:48]: Thanks, assigned you to a PR

File: .github/workflows/release.yml
Comment (pdewilde) [reply .github/workflows/release.yml:1]: Removed

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -1,52 +1,205 @@
...
+        service_url: '${{ vars.TOKEN_MINTER_SERVICE_URL }}'
+        requested_permissions: |-
+          {
+            "repositories": ["${{ github.event.repository.name }}"],
+            "permissions": {
+              "contents": "write"
+            }
+          }
+
+    - name: 'Publish release'
```
Comment (sethvargo) [inline .github/workflows/release.yml:159]: Are we not publishing binaries?

**Review by sethvargo — APPROVED** [review]

File: .github/workflows/release.yml (hunk 3 above)
Comment (pdewilde) [reply .github/workflows/release.yml:159]: No, this is a library.

Future versions might depending on where we decide to put the server part.

**Pull request merged**